### PR TITLE
clean up error messages

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: pkg/cloudprovider/cloudprovider.go [hunk 1]
```diff
@@ -66,43 +64,38 @@ func NewCloudProvider(configFile string) (c *Client, e error) {
...
 	if err != nil {
-		klog.Errorf("Get cloud env error: %+v", err)
-		return nil, err
+		return nil, fmt.Errorf("failed to get cloud env, error: %+v", err)
 	}
 
 	err = adal.AddToUserAgent(version.GetUserAgent("MIC", version.MICVersion))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to add user agent to adal for MIC, error %+v", err)
```
Comment (kkmsft) [inline pkg/cloudprovider/cloudprovider.go]: Shouldn't we use unwrap-able errors ? using %w ?

File: pkg/cloudprovider/vm.go [hunk 2]
```diff
@@ -43,7 +41,6 @@ func NewVirtualMachinesClient(config config.AzureConfig, spt *adal.ServicePrinci
 
 	reporter, err := metrics.NewReporter()
 	if err != nil {
```
Comment (kkmsft) [inline pkg/cloudprovider/vm.go]: Are the error here identifiable enough ? Should we wrap up the error to indicate where the issue was ?

File: pkg/cloudprovider/vm.go [hunk 3]
```diff
@@ -102,7 +97,6 @@ func (c *VMClient) Get(rgName string, nodeName string) (compute.VirtualMachine,
 
 	vm, err := c.client.Get(ctx, rgName, nodeName, "")
 	if err != nil {
```
Comment (kkmsft) [inline pkg/cloudprovider/vm.go]: Please confirm that the error has indications whether it happened from Get or WaitForCompletionRef .

File: pkg/cloudprovider/vmss.go [hunk 4]
```diff
@@ -43,7 +41,6 @@ func NewVMSSClient(config config.AzureConfig, spt *adal.ServicePrincipalToken) (
 
 	reporter, err := metrics.NewReporter()
 	if err != nil {
-		klog.Errorf("New reporter error: %+v", err)
 		return nil, err
```
Comment (kkmsft) [inline pkg/cloudprovider/vmss.go]: Same here: please confirm that error is identifiable.

File: pkg/metrics/prometheus_exporter.go [hunk 5]
```diff
@@ -17,18 +17,17 @@ func newPrometheusExporter(namespace string, portNumber string) (*prometheus.Exp
...
 	}
-	klog.Info("Starting Prometheus exporter")
 	// Run the Prometheus exporter as a scrape endpoint.
 	go func() {
 		mux := http.NewServeMux()
 		mux.Handle("/metrics", prometheusExporter)
 		address := fmt.Sprintf(":%v", portNumber)
 		if err := http.ListenAndServe(address, mux); err != nil {
-			klog.Errorf("Failed to run Prometheus scrape endpoint: %v", err)
+			klog.Errorf("failed to run prometheus scrape endpoint, error: %v", err)
```
Comment (kkmsft) [inline pkg/metrics/prometheus_exporter.go]: Do we need to do small caps for klog messages as well not just errors ?

**Comment by aramase** [comment]
Closing this PR! Will open a new PR with the changes.

**Pull request closed without merging**